---
title: ""
author: ""
date: ""
output:
  html_document:
    toc: true
    toc_float: true
    collapsed: false
    number_sections: false
    toc_depth: 3
    smooth_scroll: false
    theme: cerulean
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(reticulate)
library(knitr)
library(reshape2)
library(tidyverse)
library(rtracklayer)
library(ggrepel)
#
library(DiffBind)
library(TxDb.Hsapiens.UCSC.hg38.knownGene)
library(org.Hs.eg.db)
library(ChIPseeker)
library(pals)
library(reactable)
options(ChIPseeker.ignore_1st_exon = TRUE)
options(ChIPseeker.ignore_1st_intron = TRUE)
options(ChIPseeker.ignore_downstream = TRUE)
options(ChIPseeker.ignore_promoter_subcategory = TRUE)
```

# NSD2OE-NSD1KO

## Clustering
```{r echo = FALSE, warning=FALSE,message=FALSE,out.width='100%'}
setwd("~/Documents/hnscc_chipseq")
load("data/clusters_cal27_nsd2oe_v_wt.RData")
clusters_cal27_nsd2oe_v_wt$Cal27
```

```{r echo = FALSE, warning=FALSE,message=FALSE,out.width='100%'}
load('data/cons_nsd2oe.rda')
g <- import.bed('data/gene.bed')
ig <- import.bed('data/intergenic.bed')
A_genes <- subsetByOverlaps(cons$A,g)
B_genes <- subsetByOverlaps(cons$B,g)
A_df <- data.frame("Cluster"=c("A","B"),
                   "Percent_Genic"=c("36.7","52.5"),
                   "Percent_Intergenic"=c("63.3","47.5"))
reactable(A_df)
```

## Annotation
```{r include=FALSE}
listCons <- as.list(cons)
annot <- lapply(cons,function(x){annotatePeak(x,tssRegion = c(-3000, 3000),
             TxDb = TxDb.Hsapiens.UCSC.hg38.knownGene,
             annoDb = "org.Hs.eg.db")})
```

```{r echo = FALSE, warning=FALSE,message=FALSE,out.width='100%'}
plotAnnoBar(annot)
```

## Ensembl Enrichment {.tabset}

Nonetheless, we can proceed with the representative bins in those clusters we identified and examine whether they preferentially overlap certain elements.

Here we see consistent enrichment in CTCF binding sites.

Recently, NSD2-overexpression has been shown to alter CTCF binding within K36me2 enriched regions. [https://www.nature.com/articles/s41467-019-12811-4](DOI: 10.1038/s41467-019-12811-4). Similarly, regions that had increased K36me2 in NSD2OE relative to parental seems to be enriched at intragenic CTCF binding sites (cluster A). However, in intergenic regions, the effect is more indiscriminate and not specific to an annotated region, although some of the bins also overlap with intergenic CTCF binding sites. 

### Genome-wide
```{r, fig.cap="Overlap enrichment result of Ensembl annotations. The size of the dots corresponds to number of bins overlapping the corresponding annotation. *** represents p-value < 1e-5 based on Fisher’s exact test of bins overlapping a specific class of annotated regions versus a background of all non-quiescent bins, meaning >10 reads in at least one mark in one sample.",echo = FALSE, warning=FALSE,message=FALSE,out.width='100%'}
load("data/or_cal27_nsd2oe_v_wt.RData")
or_cal27_nsd2oe_v_wt$`Genome-wide`$A
```

```{r,fig.cap="Overlap enrichment result of Ensembl annotations. The size of the dots corresponds to number of bins overlapping the corresponding annotation. *** represents p-value < 1e-5 based on Fisher’s exact test of bins overlapping a specific class of annotated regions versus a background of all non-quiescent bins, meaning >10 reads in at least one mark in one sample.",echo = FALSE, warning=FALSE,message=FALSE,out.width='100%'}
or_cal27_nsd2oe_v_wt$`Genome-wide`$B
```

### Genic
```{r echo = FALSE, warning=FALSE,message=FALSE,out.width='100%'}
or_cal27_nsd2oe_v_wt$Genic$A
```

```{r echo = FALSE, warning=FALSE,message=FALSE,out.width='100%', fig.cap="snRNA = small nuclear RNA, which are associated with small nuclear ribonucleoproteins and involved in the processing of pre-mRNA"}
or_cal27_nsd2oe_v_wt$Genic$B
```

### Intergenic

No significant overlap enrichment was found for the intergenic bins in cluster A. Hence, we will look at the # bins overlapping these annotated regions instead.

```{r echo = FALSE, warning=FALSE,message=FALSE,out.width='100%'}
load(file="~/Documents/hnscc_chipseq/data/ol_intergenic_clusterA_nsd2oe.RData")
ol_intergenic_clusterA_nsd2oe$Intergenic$A
ol_intergenic_clusterA_nsd2oe$Genic$A
```

```{r echo = FALSE, warning=FALSE,message=FALSE,out.width='100%'}
or_cal27_nsd2oe_v_wt$Intergenic$B
```


